### PR TITLE
Make "pgroonga" 0.9.0 to be optional resource in postgresql formula

### PR DESCRIPTION
**Pull request by cosmo0920**
Revised https://github.com/Homebrew/homebrew/pull/41174 with `optional resource` technique.

### CODE REVIEW

**Comment by cosmo0920** [comment]
This PR should solve sandbox issue, but test-bot CI reports tests failure.
How do I fix them?

File: Library/Formula/postgresql.rb [hunk 1]
```diff
@@ -24,15 +24,23 @@ class Postgresql < Formula
...
+  resource "pgroonga" do
+    url "http://packages.groonga.org/source/pgroonga/pgroonga-0.9.0.tar.gz"
+    sha256 "846b89d20c847bf54103978e5234fab0fbf95eafdcfffb35eac5273a26357f51"
+  end
+
   depends_on "openssl"
   depends_on "readline"
   depends_on "libxml2" if MacOS.version <= :leopard # Leopard libxml is too old
   depends_on :python => :optional
+  depends_on "groonga" => :build if build.with? "pgroonga"
```
Comment (DomT4) [inline Library/Formula/postgresql.rb]: If two variables are the same block them:

```ruby
if build.with? "pgroonga"
  depends_on "groonga" => :build
  depends_on "pkg-config" => :build
end
```

File: Library/Formula/postgresql.rb [hunk 2]
```diff
@@ -24,15 +24,23 @@ class Postgresql < Formula
   option "without-perl", "Build without Perl support"
   option "without-tcl", "Build without Tcl support"
   option "with-dtrace", "Build with DTrace support"
+  option "with-pgroonga", "Build with the PGroonga postgresql extension."
 
   deprecated_option "no-perl" => "without-perl"
   deprecated_option "no-tcl" => "without-tcl"
   deprecated_option "enable-dtrace" => "with-dtrace"
 
+  resource "pgroonga" do
```
Comment (DomT4) [inline Library/Formula/postgresql.rb]: Nit but resources should go beneath dependencies.

File: Library/Formula/postgresql.rb [hunk 3]
```diff
@@ -82,6 +90,15 @@ def install
 
     system "./configure", *args
     system "make", "install-world"
+
+    # Now build and install pgroonga as optional resource
+    if build.with? "pgroonga"
+      resource("pgroonga").stage do
+        ENV.append_path "PATH", "#{Formula["postgresql"].bin}"
```
Comment (DomT4) [inline Library/Formula/postgresql.rb]: Doesn't need the surrounding `""` here. Should work as `ENV.append_path "PATH", Formula["postgresql"].bin`

**Comment by DomT4** [comment]
> This PR should solve sandbox issue, but test-bot CI reports tests failure.
How do I fix them?

On El Capitan it's using the deprecated system OpenSSL instead of ours. It probably needs something like `--with-openssl=#{Formula["openssl"].opt_bin}.` Check the `configure` switches and see what's available on that.

The other failure is a sandbox failure elsewhere, with the test trying to use a configuration file we no longer permit:

```
Testing pgpool-ii
==> Using the sandbox
==> /usr/local/Cellar/pgpool-ii/3.4.2/bin/pg_md5 --md5auth pool_passwd --config-file /usr/local/etc/pgpool.conf.sample
ERROR: pid 81709: initializing pool password, failed to open file:"/usr/local/etc/pool_passwd"
Error: pgpool-ii: failed
Failed executing: /usr/local/Cellar/pgpool-ii/3.4.2/bin/pg_md5 --md5auth pool_passwd --config-file /usr/local/etc/pgpool.conf.sample
```

**Comment by cosmo0920** [comment]
> Check the configure switches and see what's available on that.

`--with-openssl` option is available, but it does not nothing passed there.
I fixed to pass `Formula["openssl"].opt_prefix` into this option.

File: Library/Formula/postgresql.rb [hunk 4]
```diff
@@ -82,6 +93,15 @@ def install
 
     system "./configure", *args
     system "make", "install-world"
+
+    # Now build and install pgroonga as optional resource
+    if build.with? "pgroonga"
+      resource("pgroonga").stage do
+        ENV.append_path "PATH", Formula["postgresql"].bin
```
Comment (xu-cheng) [inline Library/Formula/postgresql.rb]: It should be just `bin` here. You don't need `Formula["postgresql"]` to point itself.

**Comment by xu-cheng** [comment]
Can you add a test for this option? Something like:
```ruby
test do
  ...
  if build.with? "pgroonga"
    ...
  end
end
```

Also our CI won't test non default option. So you should test this locally.

**Comment by DomT4** [comment]
> Also our CI won't test non default option. 

I'm not sure it's worth adding tests for non-default options. We've refused to do this several times over the last couple months even because it opens the floodgates to having a test element for every option even though we'd never use it. Has our position changed here?

**Comment by xu-cheng** [comment]
> We've refused to do this several times over the last couple months

I was unaware this. Please ignore my comment on adding new test. But still this option need to be tested locally.

**Comment by DomT4** [comment]
I'm fine if we want to have that team discussion; just didn't know if I'd missed something somewhere :)

**Comment by cosmo0920** [comment]
>>    Check the configure switches and see what's available on that.

> --with-openssl option is available, but it does not nothing passed there.
> I fixed to pass Formula["openssl"].opt_prefix into this option.

Hmm, `--with-openssl` does not accept arguments.... :/

File: Library/Formula/postgresql.rb (hunk 4 above)
Comment (cosmo0920) [inline Library/Formula/postgresql.rb]: I've confirmed that building `pgroonga` succeeded using `bin` here.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
This is passing the CI now. The OpenSSL issue got fixed with the version bumps & changes merged yesterday. How does `pgroonga` know where to install, out of interest? I noticed there's no prefix argument being passed anywhere.

**Comment by cosmo0920** [comment]
> How does pgroonga know where to install, out of interest? I noticed there's no prefix argument being passed anywhere.

Ah..., yes. Is this not suitable for Homebrew's policy?
`PGroonga`'s makefile calculates suitable installation location.
In this situation, `pgroonga.so` is installed in `$(brew -prefix)/Cellar/postgresql/<version>/lib` and `pgroonga.sql` is installed in  `$(brew -prefix)/Cellar/postgresql/<version>/share/postgresql/extension/`.

**Comment by DomT4** [comment]
It's fine if that's the case. Just wanted to check it was installing into the prefix and not a generic `/usr/local` directory or such.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/022faf82010f19e8eeac36ba4079ce266f4b2b8d. Thank you for another contribution to Homebrew @cosmo0920 :star2:

**Comment by cosmo0920** [comment]
Thanks! I'm appreciate to make to be pgroonga optional resource in postgresql formula!

**Pull request closed without merging**